test/run_pass/test_net_http_outgoing_buffer.js

```diff
@@ -0,0 +1,42 @@
+/* Copyright 2015-present Samsung Electronics Co., Ltd. and other contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var net = require('net');
+var http = require('http');
+var fs = require('fs');
+var assert = require('assert');
+
+var binaryMessage = fs.readFileSync('./resources/random_binarydata');
+assert.notEqual(binaryMessage.length, 0);
+
+var server = http.createServer(function(req, res) {
+  req.on('data', function(data) {
+    assert.equal(data.compare(binaryMessage), 0);
+    res.end();
+  });
+}).listen(8383, 5);
+
+var reqOpts = {
+  method: 'POST',
+  port: 8383,
+  path: '/',
+  headers: {'Content-Length': binaryMessage.length},
+};
+
+var clientReq = http.request(reqOpts, function(res) {
+  server.close();
+});
+
+clientReq.end(binaryMessage);
```

Comment:
We should also test the other way round. That is the server sends a binary data and the client processes it.

Reply:
It calls the exact same function so I decided not to.
However I could add it as well.